src/Eccube/Controller/Admin/Order/README.html

```
@@ -0,0 +1,88 @@
+
+<section data-section="notes" data-customer="false">
+ <h2>注意点<span class="dev-only">開発者向け</span></h2>
+ <ul>
+  <li><strong>管理操作が実世界を変える「二重副作用」に注意。</strong> <code>EditController</code> / <code>ShippingController</code> は管理画面の操作から <code>PurchaseFlow.prepare()</code>/<code>commit()</code> を呼ぶため、在庫引当・注文番号採番・（設定によっては）メール送信が管理画面から発火する。確定系を安易に増やさない。</li>
+  <li><strong>受注ステータスは <code>OrderStateMachine</code> 経由で遷移させる。</strong> 直接 <code>setOrderStatus()</code> で書き換えず、遷移可否チェックを通す。</li>
+  <li><strong>Fat コントローラを増やさない。</strong> 受注の計算・検証・確定は PurchaseFlow へ。コントローラは HTTP 入出力と画面遷移に徹する。</li>
+  <li><strong>認可・CSRF。</strong> パスは <code>%eccube_admin_route%</code> 配下（admin ファイアウォールで保護）。一括削除・ステータス更新・伝票番号更新など GET 以外の状態変更は CSRF トークンを検証する。</li>
+  <li><strong>ルートの <code>{id}</code> が指す対象に注意。</strong> 出荷編集 <code>admin_shipping_edit</code>（<code>/shipping/{id}/edit</code>）の <code>{id}</code> は <strong>Shipping ではなく Order の ID</strong>で、1 受注の全お届け先をまとめて編集する。一方で対応状況・送り状番号の更新（<code>admin_shipping_update_order_status</code> / <code>admin_shipping_update_tracking_number</code>）は <code>ShippingController</code> ではなく <code>OrderController</code> に置かれ、<code>{id}</code> は Shipping の ID を指す。</li>
+  <li><strong>発送済（<code>DELIVERED</code>）遷移は全出荷が揃って初めて受注に反映。</strong> 出荷 1 件を発送済にしても、受注ステータスが <code>DELIVERED</code> へ進むのは <strong>受注内の全 Shipping が発送済のときだけ</strong>。未発送の出荷が残る間は当該 Shipping の <code>shippingDate</code> だけ設定される（部分出荷）。</li>
+  <li><strong>ステータス変更が在庫・会員サマリを動かす。</strong> <code>IN_PROGRESS</code>／<code>CANCEL</code> への遷移は商品在庫を増減し、受注に会員が紐づく場合はいずれの遷移でも <code>updateOrderSummary()</code> で購入回数・購入金額を再計算する（受注登録・編集でも同様）。</li>
+  <li><strong>一括削除は物理削除。</strong> <code>bulkDelete()</code> は <code>$em->remove($Order)</code> による物理削除で、論理削除フラグを立てるのではなく受注そのものを消す。</li>
```

HTMLHint (1.9.2)

[error] Special characters must be escaped : [ > ].

(spec-char-escape)